src/Migration/MessageQueue/Handler/ResetChecksumHandler.php

```diff
@@ -0,0 +1,228 @@
+            ->select('m.id')
+            ->from('swag_migration_mapping', 'm')
+            ->where('m.checksum IS NOT NULL')
+            ->andWhere('m.connection_id = :connectionId')
+            ->setParameter('connectionId', Uuid::fromHexToBytes($connectionId))
+            ->setMaxResults(self::BATCH_SIZE);
+
+        if (!$message->isResettingAll()) {
+            $query->innerJoin(
+                'm',
+                'swag_migration_data',
+                'd',
+                'd.mapping_uuid = m.id AND d.written = 0'
+            );
+        }
+
+        $ids = $query->executeQuery()->fetchFirstColumn();
+        $batchSize = \count($ids);
+
+        if ($batchSize === 0) {
+            $this->clearResettingChecksumsFlag();
+            $this->handleCompletion($message, $progress);
+
+            return;
+        }
+
+        $this->connection->createQueryBuilder()
+            ->update('swag_migration_mapping')
+            ->set('checksum', 'NULL')
+            ->where('id IN (:ids)')
+            ->setParameter('ids', $ids, ArrayParameterType::BINARY)
+            ->executeStatement();
+
+        $newProcessedCount = $message->getProcessedMappings() + $batchSize;
+
+        if ($message->getRunId() !== null) {
+            $progress = $this->updateProgress(
+                $message,
+                $newProcessedCount,
+                $totalMappings,
+                $message->getContext()
+            );
+        }
+
+        if ($batchSize < self::BATCH_SIZE) {
+            $this->clearResettingChecksumsFlag();
+            $this->handleCompletion($message, $progress);
+
+            return;
+        }
+
+        $this->messageBus->dispatch(new ResetChecksumMessage(
+            $message->getConnectionId(),
+            $message->getContext(),
+            $message->isResettingAll(),
+            $message->getRunId(),
+            $message->getEntity(),
+            $totalMappings,
+            $newProcessedCount,
+            $message->isPartOfAbort()
+        ));
+    }
+
+    private function handleCompletion(ResetChecksumMessage $message, ?MigrationProgress $progress): void
+    {
+        if (!$message->isPartOfAbort() || $message->getRunId() === null) {
+            return;
+        }
+
+        $runId = $message->getRunId();
+        $context = $message->getContext();
+
+        $this->runTransitionService->forceTransitionToRunStep(
+            $runId,
+            MigrationStep::CLEANUP
+        );
+
+        $finalProgress = new MigrationProgress(
+            0,
+            0,
+            $progress?->getDataSets() ?? new ProgressDataSetCollection(),
+            $message->getEntity() ?? DefaultEntities::RULE,
+            $progress?->getCurrentEntityProgress() ?? 0
+        );
+        $finalProgress->setIsAborted(true);
+
+        $this->migrationRunRepo->upsert([
+            [
+                'id' => $runId,
+                'progress' => $finalProgress,
```

Comment:
In the other places you execute the update like this:

```suggestion
                'progress' => $finalProgress->jsonSerialize(),
```